From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ömer Faruk Nehir <92220159+omerfaruknehir@users.noreply.github.com>
Date: Thu, 13 Aug 2026 23:31:00 +0300
Subject: [PATCH] arm64: dts: qcom: x1-asus-zenbook-a14: add OV02C10 EEPROM

Describe the 8 KiB M24C64-family EEPROM on CCI1/I2C1 as a read-only
at24 device at Linux I2C address 0x50. This exposes its contents through
the standard NVMEM interface.

Signed-off-by: Ömer Faruk Nehir <92220159+omerfaruknehir@users.noreply.github.com>
---
 arch/arm64/boot/dts/qcom/x1-asus-zenbook-a14.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/x1-asus-zenbook-a14.dtsi b/arch/arm64/boot/dts/qcom/x1-asus-zenbook-a14.dtsi
--- a/arch/arm64/boot/dts/qcom/x1-asus-zenbook-a14.dtsi
+++ b/arch/arm64/boot/dts/qcom/x1-asus-zenbook-a14.dtsi
@@ -878,6 +878,16 @@ &cci1_i2c1 {
 			};
 		};
 	};
+
+	ov02c10_eeprom: eeprom@50 {
+		compatible = "st,24c64", "atmel,24c64";
+		reg = <0x50>;
+		vcc-supply = <&vreg_l3m_1p8>;
+		read-only;
+		address-width = <16>;
+		size = <8192>;
+	};
 };
 
 &csiphy4 {
-- 
2.53.0
